app/src/main/java/com/example/cachua/musicdemo21/service/MyMusicService.kt

```diff
@@ -0,0 +1,242 @@
+    }
+
+
+    inner class playMusicBinder : Binder() {
+        fun service(): MyMusicService {
+            return this@MyMusicService
+        }
+    }
+
+
+    fun isPlaying(): Int {
+        return mStatusService
+    }
+
+    fun onProgress(i: Int) {
+        mMediaPlayer.seekTo(i * mMusic.duration / 100)
+    }
+
+
+    fun selectSong(mContext: Context, mArrayListMusic: ArrayList<MusicModel>, position: Int) {
+        this.mArrayListMusic = mArrayListMusic
+        mPosition = position
+        mMediaPlayer.reset()
+        mMediaPlayer.setDataSource(mContext, Uri.parse(mArrayListMusic[position].data))
+        mMediaPlayer.prepare()
+        mMusic = mArrayListMusic[position]
+
+        if (mStatusService == 0) {
+            mMediaPlayer.start()
+        }
+        upDateMusicTime(mMusic, 0, mPosition)
+    }
+
+    fun nextSong() {
+        mPosition = if (mPosition < mArrayListMusic.size - 1) {
+            mPosition + 1
+        } else {
+            0
+        }
+        mMediaPlayer.reset()
+        mMediaPlayer.setDataSource(this, Uri.parse(mArrayListMusic[mPosition].data))
+        mMediaPlayer.prepare()
+        mMusic = mArrayListMusic[mPosition]
+        if (mStatusService == 0) {
+            mMediaPlayer.start()
+        }
+        upDateMusicTime(mMusic, 0, mPosition)
+    }
+
+    fun previousSong() {
+        mPosition = if (mPosition <= 0) {
+            mArrayListMusic.size - 1
+        } else {
+            mPosition - 1
+        }
+        mMediaPlayer.reset()
+        mMediaPlayer.setDataSource(this, Uri.parse(mArrayListMusic[mPosition].data))
+        mMediaPlayer.prepare()
+        mMusic = mArrayListMusic[mPosition]
+        if (mStatusService == 0) {
+            mMediaPlayer.start()
+        }
+        upDateMusicTime(mMusic, 0, mPosition)
+
+    }
+
+    fun playSong() {
+        mMediaPlayer.start()
+        mStatusService = 0
+        upDateMusicTime(mMusic, mMediaPlayer.currentPosition, mPosition)
+        pushNotification(mMusic)
+    }
+
+    fun pauseSong() {
+        mMediaPlayer.pause()
+        mStatusService = 1
+        pushNotification(mMusic)
+    }
+
+    fun upDateMusicTime(music: MusicModel, progress: Int, position: Int) {
+        Handler().postDelayed({
+            if (progress == 0) {
+                pushNotification(music)
+            }
+            if (progress == 0 && mStatusService == 1) {
+                mCallBack.onUpTime(music, mMediaPlayer.currentPosition)
+            }
+            if(mStatusService == 0 && !mMediaPlayer.isPlaying){
+                nextSong()
+            }
+            if (mStatusService == 0 && position == mPosition) {
+                mCallBack.onUpTime(music, mMediaPlayer.currentPosition)
+                upDateMusicTime(music, mMediaPlayer.currentPosition, position)
+            }
+        }, 1000)
+    }
+
+
+    fun pushNotification(music: MusicModel) {
+        var intentNotification = Intent(this, MusicActivity::class.java).apply {
```

Comment:
```suggestion
        val intentNotification = MainActivity.getIntent(this).apply {
```
Bên MainActivity sẽ thêm đoạn
```kotlin
         companion object {
              fun getIntent(context: Context) = Intent(context, MainActivity::class.java)
         }
```

Reply:
ý anh Đạt là ntn nhé: https://github.com/framgia/coding-standards/blob/master/vn/android/codingstyleguide.md#211-%C4%91%E1%BB%91i-s%E1%BB%91-trong-activity-v%C3%A0-fragment
```suggestion
        companion object {
              @JvmStatic
              fun getIntent(context: Context) = Intent(context, MainActivity::class.java)
         }
```
như vậy bên service chỉ cần gọi như anh Đạt chỉ.